node/lib/cmd/submodule.js

```
@@ -135,3 +150,126 @@ exports.executeableSubcommand = co.wrap(function *(args) {
+    // Here, we find which submodule `path` refers too.  It migt be invalid by
+    // referring to no submodule, or by referring to more than one.
+
+    const relPath = yield GitUtil.resolveRelativePath(workdir,
+                                                      process.cwd(),
+                                                      path);
+    const resolvedPaths = yield SubmoduleUtil.resolvePaths(workdir,
+                                                           [relPath],
+                                                           subNames,
+                                                           openSubNames);
+    const paths = Object.keys(resolvedPaths);
+    if (0 === paths.length) {
+        throw new UserError(`No submodule found in ${colors.red(path)}.`);
+    }
+    if (1 !== paths.length) {
+        throw new UserError(`Multiple submodules: \
+${colors.yellow(paths.join(", "))} found in ${colors.red(path)}.`);
+    }
+
+    // Create a `SubmoduleFetcher` to be used for opening the submodule, if
+    // closed, and for fetching commits during the search.
+
+    const head = yield repo.getHeadCommit();
+    const fetcher = new SubmoduleFetcher(repo, head);
+
+    const subName = paths[0];
+    let subRepo;
+
+    // Open the submodule if it's closed.
+
+    if (-1 === openSubNames.findIndex(x => x === subName)) {
+        console.log(`Opening ${colors.blue(subName)}.`);
+        const shas = yield SubmoduleUtil.getSubmoduleShasForCommit(repo,
+                                                                   [subName],
+                                                                   head);
+        const subSha = shas[subName];
+        subRepo = yield Open.openOnCommit(fetcher, subName, subSha);
+    }
+    else {
+        subRepo = yield SubmoduleUtil.getRepo(repo, subName);
+    }
+
+    // Now we resolve the committishes.
+
+    const metaAnnotated = yield GitUtil.resolveCommitish(repo, metaCommittish);
+    if (null === metaAnnotated) {
```

We could probably fail faster than this! Especially since it'll open the sub-repo as a side effect.